javav2/example_code/redshift/src/main/java/com/example/scenario/RedshiftScenario.java

```diff
@@ -0,0 +1,550 @@
+        System.out.println("The identifier of the statement is " + id);
+        checkStatement(redshiftDataClient, id);
+        getResults(redshiftDataClient, id);
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("Now you will list your databases.");
+        System.out.print("Press Enter to continue...");
+        scanner.nextLine();
+        listAllDatabases(redshiftDataClient, clusterId, userName, databaseName);
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("Now you will modify the Redshift cluster.");
+        System.out.print("Press Enter to continue...");
+        scanner.nextLine();
+        modifyCluster(redshiftClient, clusterId);
+        System.out.println(DASHES);
+
+
+        System.out.println(DASHES);
+        System.out.println("Would you like to delete the Amazon Redshift cluster? (y/n)");
+        String delAns = scanner.nextLine().trim();
+        if (delAns.equalsIgnoreCase("y")) {
+            System.out.println("You selected to delete " +clusterId);
+            System.out.print("Press Enter to continue...");
+            scanner.nextLine();
+            deleteRedshiftCluster(redshiftClient, clusterId);
+        } else {
+            System.out.println("The "+clusterId +" was not deleted");
+        }
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("This concludes the Amazon Redshift example scenario.");
+        System.out.println(DASHES);
+    }
+
+    public static void listAllDatabases(RedshiftDataClient redshiftDataClient, String clusterId, String dbUser, String database) {
+        try {
+            ListDatabasesRequest databasesRequest = ListDatabasesRequest.builder()
+                .clusterIdentifier(clusterId)
+                .dbUser(dbUser)
+                .database(database)
+                .build();
+
+            ListDatabasesIterable listDatabasesIterable = redshiftDataClient.listDatabasesPaginator(databasesRequest);
+            listDatabasesIterable.stream()
+                .flatMap(r -> r.databases().stream())
+                .forEach(db -> System.out
+                    .println("The database name is : " + db));
+
+        } catch (RedshiftDataException e) {
+            System.err.println(e.getMessage());
+            System.exit(1);
+        }
+    }
+
+    // snippet-start:[redshift.java2.delete_cluster.main]
+    public static void deleteRedshiftCluster(RedshiftClient redshiftClient, String clusterId) {
+        try {
+            DeleteClusterRequest deleteClusterRequest = DeleteClusterRequest.builder()
+                .clusterIdentifier(clusterId)
+                .skipFinalClusterSnapshot(true)
+                .build();
+
+            DeleteClusterResponse response = redshiftClient.deleteCluster(deleteClusterRequest);
+            System.out.println("The status is " + response.cluster().clusterStatus());
```

Comment:
I get an output saying the status is "deleting". Shouldn't the scenario wait until the delete is finished before completing?

Reply:
We could loop until its fully deleted.